client/my-sites/migrate/section-migrate.js

```diff
@@ -39,36 +68,166 @@ class SectionMigrate extends Component {
 		} );
 	};
 
-	render() {
+	startMigration = () => {
+		const { sourceSiteId, targetSiteId } = this.props;
+
+		if ( ! sourceSiteId || ! targetSiteId ) {
+			return;
+		}
+
+		this.setState( { migrationStatus: 'restoring' } );
+		wpcom.startMigration( sourceSiteId, targetSiteId ).then( () => this.updateFromAPI() );
+	};
+
+	updateFromAPI = () => {
+		const { targetSiteId } = this.props;
+		wpcom
+			.getMigrationStatus( targetSiteId )
+			.then( response => {
+				const { status: migrationStatus } = response;
+				if ( migrationStatus ) {
+					this.setState( { migrationStatus } );
+				}
+			} )
+			.catch( () => {
+				// @TODO: handle status error
+			} );
+	};
+
+	renderLoading() {
+		return (
+			<CompactCard>
+				<span className="migrate__placeholder">Loading...</span>
+			</CompactCard>
+		);
+	}
+
+	renderMigrationComplete() {
+		const { targetSite } = this.props;
+		const viewSiteURL = get( targetSite, 'URL' );
+
+		return (
+			<CompactCard>
+				<div className="migrate__status">Your migration has completed successfully.</div>
+				<Button primary href={ viewSiteURL }>
+					View site
+				</Button>
+				<Button onClick={ this.resetMigration }>Start over</Button>
```

Comment:
This seems a bit 'dangerous', but I'm cool to keep it in there for now since this is dev-only.